src/agent.ts

```diff
@@ -57,15 +58,35 @@ export class Agent {
       // Do not act again if the last event was a decision
       if (last(events)?.type === "decision") return;
 
+      const t0 = Date.now();
+      this.messageBus.send(
+        messageBuilder.agentToAgent(
+          this.id,
+          [AGENT_ID],
+          `Reflecting on ${events.length} events...`
+        )
+      );
+
       const actionText = await makeDecision(events);
 
+      this.messageBus.send(
+        messageBuilder.agentToAgent(
+          this.id,
+          [AGENT_ID],
+          `Arrived at a decision after ${((Date.now() - t0) / 1000).toFixed(
+            1
+          )}s  \n\n ${actionText}`
+        )
+      );
+
```

Comment:
I don't think we need to send these as messages to the control agent. If the user is interested in when the agent is thinking they will presumably be using your UI to monitor it.